dgl_sparse/src/matmul.cc

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright (c) 2022 by Contributors
+ * @file matmul.cc
+ * @brief DGL sparse matrix multiplication functions.
+ */
+#include "./matmul.h"
+
+// clang-format off
+#include <sparse/dgl_headers.h>
+// clang-format on
+
+#include <sparse/sparse_matrix.h>
+#include <torch/script.h>
+
+#include "./utils.h"
+
+namespace dgl {
+namespace sparse {
+
+torch::Tensor SpMMNoAutoGrad(
+    const c10::intrusive_ptr<SparseMatrix>& sparse_mat,
+    torch::Tensor sparse_val, torch::Tensor dense_mat, bool transpose_sparse) {
+  const std::string op = "mul";
+  const std::string reduce = "sum";
+  const int64_t out_row =
+      transpose_sparse ? sparse_mat->shape()[1] : sparse_mat->shape()[0];
+  const std::vector<int64_t> shape = {out_row, dense_mat.size(1)};
+
+  auto ret = torch::zeros(shape, dense_mat.options());
+  auto dgl_sparse_val = TorchTensorToDGLArray(sparse_val);
+  auto dgl_dense_mat = TorchTensorToDGLArray(dense_mat);
+  auto dgl_ret = TorchTensorToDGLArray(ret);
+  if (!transpose_sparse) {
+    if (sparse_mat->HasCSR() || !sparse_mat->HasCOO()) {
+      auto csr = CSRToOldDGLCSR(sparse_mat->CSRPtr());
```

Comment:
As discussed, please add comment here.
// sparse_mat->CSRPtr() will implicitly convert CSC to CSR format.